packages/@lwc/ssr-runtime/src/render.ts

```diff
@@ -39,22 +58,80 @@ export function* fallbackTmpl(
     }
 }
 
+export function fallbackTmplNoYield(
+    emit: (segment: string) => void,
+    _props: unknown,
+    _attrs: unknown,
+    _slotted: unknown,
+    Cmp: LightningElementConstructor,
+    _instance: unknown
+) {
+    if (Cmp.renderMode !== 'light') {
+        emit('<template shadowrootmode="open"></template>');
+    }
+}
+
 export type GenerateMarkupFn = (
     tagName: string,
     props: Record<string, any> | null,
     attrs: Attributes | null,
     slotted: Record<number | string, AsyncGenerator<string>> | null
 ) => AsyncGenerator<string>;
 
+export type GenerateMarkupFnAsyncNoGen = (
+    emit: (segment: string) => void,
+    tagName: string,
+    props: Record<string, any> | null,
+    attrs: Attributes | null,
+    slotted: Record<number | string, AsyncGenerator<string>> | null
```

Comment:
The number of arguments that a `GenerateMarkupFn` might expect will probably change over time. The same goes for the `tmpl` functions, which also need to accept the new `$$emit` argument. Because `generateMarkup` and `tmpl` are generated code, it seemed cleaner/safer to transform `yield generateMarkup(/* some number of args */)` into `generateMarkup($$emit, /* some number of args */)`.

That said, this is just a stylistic preference. I think the post-transmogrified code reads better this way. But if y'all feel strongly, I can make the change.

Reply:
It's a taste thing, but I prefer `$$emit` as the first arg as well.